pkg/operator/ceph/object/s3-handlers.go

```diff
@@ -36,7 +36,7 @@ type S3Agent struct {
 	Client *s3.S3
 }
 
-func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byte) (*S3Agent, error) {
+func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byte, insecure bool) (*S3Agent, error) {
```

Comment:
Instead of a bool, I'd suggest we actually have two different methods so so don't accidentally pass the wrong flag in the production code. We would have:
```go
func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byte) (*S3Agent, error) {
  return newS3Agent(..., false)
}
func NewTestOnlyS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byte) (*S3Agent, error) {
  return newS3Agent(..., true)
}
func newS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byte, insecure bool) (*S3Agent, error) {
  // implementation
})
```